src/methods/wallet-request-execution-permissions.ts

```diff
@@ -0,0 +1,89 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { Infer } from '@metamask/superstruct';
+import {
+  array,
+  boolean,
+  literal,
+  object,
+  optional,
+  record,
+  string,
+  union,
+  unknown,
+} from '@metamask/superstruct';
+import {
+  HexChecksumAddressStruct,
+  type Hex,
+  type Json,
+  type JsonRpcRequest,
+  type PendingJsonRpcResponse,
+  StrictHexStruct,
+} from '@metamask/utils';
+
+import { validateParams } from '../utils/validation';
+
+const PermissionStruct = object({
+  type: string(),
```

Comment:
do we want to make this an enum and enforce it at this level of the middleware, or is that too high up in the stack / not the primary concern of this middleware?